### PR TITLE
Improve message for untyped module with noImplicitAny

### PR DESCRIPTION
**Pull request by sandersn**
Add a suggestion to use `npm install @types/foo` or `declare module 'foo'` in a separate file.
As requested by @bterlson to ease the new user experience.

Not sure about Markdown in error messages. @RyanCavanaugh what kind of punctuation is best to use to set off code examples?

### CODE REVIEW

File: src/compiler/diagnosticMessages.json [hunk 1]
```diff
@@ -3310,6 +3310,10 @@
         "category": "Error",
         "code": 7034
     },
+    "Try `npm install @types/{0}` if it exists or adding `declare module '{0}'` in a separate file.": {
```
Comment (bterlson) [inline src/compiler/diagnosticMessages.json]: Is "separate file" descriptive enough? Can we go with "a global declaration (d.ts) file" or something?

I might also want a pointer as to where to put the file but I can't think of a concise way to say it ("in your compilation" is one attempt but is probably confusing for novices).

Comment (sandersn) [inline src/compiler/diagnosticMessages.json]: Technically, any old .ts file is fine. @andy-ms do we provide any guidance on the *right* way to do this?

For "in your compilation", maybe reword the last half to say "... or add a new file [to your project] containing `declare module 'foo'`." Sound better? I'm not sure about 'to your project'.

Comment (mhegazy) [inline src/compiler/diagnosticMessages.json]: I agree with @bterlson, technically any non-module file would work, but it is clearer to say `in a separate .d.ts file.`

Comment (bterlson) [inline src/compiler/diagnosticMessages.json]: I want to see both "declaration file" and ".d.ts" because "typescript declaration file" is more easily searchable and goes straight to the proper handbook topic :)

@sandersn otherwise I like your tweaks. I guess I agree that "to your project" doesn't add anything. I was trying to nod to the fact that you can't just put the .d.ts in the root of your project and expect it to work - it's gotta be somewhere tsc is configured to look. But maybe that's just obvious, and in any case "to your project" doesn't really say that.

Comment (sandersn) [inline src/compiler/diagnosticMessages.json]: I hope new users start with `excludes: ["node_modules"]` or something so that almost all files are included anyway. In any case, the tweaked version is up now.
